File: javascript/armstrong_for_n_digits.js
//working...for any no. of digits..
//success..
const readline=require('readline').createInterface({
	input: process.stdin,
	output: process.stdout
})
console.log('we will check armstrong for a number of any digits..')
readline.question('enter a number\n',user=>{
	check_armstrong(user);
	readline.close();
})
function check_armstrong(n) {
	let copy=n;
	const len=n.length;
	let digit,result=0;
	while( copy>0){
		digit=copy%10;
		result+=(digit**len);
		copy=parseInt(copy/10);
	}
	if(result==n)
		console.log(`${n} is an armstrong number`);
	else
		console.log(`${n} is not an armstrong number`);
}

File: java/armstrong_for_n_digits.java
import java.util.*;//working.......for n no. of digits..
public class armstrong_for_n_digits{//success.....
	public static void main(String k[]){
		Scanner nrt=new Scanner(System.in);
		System.out.println("we will check armstrong number for n digits\nenter a number");
		int n=nrt.nextInt();
		int copy1=n;
		int copy2;
		int result=0;
		int no_digits=0,digit;
		for(;copy1!=0;copy1/=10){
            ++no_digits;
		}
		copy2=n;
		for(;copy2!=0;copy2/=10){
			digit=copy2%10;
			result+=Math.pow(digit,no_digits);
		}
		if(result==n)
			System.out.println(n+" is an armstrong number");
		else
			System.out.println(n+" is not an armstrong Number");
		nrt.close();
	}
}